internal/buffer/line_array.go

```diff
@@ -337,27 +337,39 @@ func (la *LineArray) LineBytes(lineN int) []byte {
 func (la *LineArray) State(lineN int) highlight.State {
 	la.lines[lineN].lock.Lock()
 	defer la.lines[lineN].lock.Unlock()
+	if lineN >= len(la.lines) || lineN < 0 {
```

Comment:
> These should be before trying to lock `la.lines[lineN].lock`.
> 
> Because of this, the test actually still fails: I've run the test 100 times, 99 passed, but 1 failed, with `index out of range` in `SetState()`.

Damn, I watched a dozens of times in the evening at this location, but now I realize. Obviously it was too late again.
Thank you!